Dockerfile

```diff
@@ -140,17 +140,15 @@ ARG PYTHON_INSPECTOR_VERSION=0.9.2
 ARG PYTHON_PIPENV_VERSION=2022.9.24
 ARG PYTHON_POETRY_VERSION=1.2.2
 ARG PIPTOOL_VERSION=22.2.2
-ARG SCANCODE_VERSION=30.1.0
+ARG SCANCODE_VERSION=31.2.1
```

Comment:
If you want to continue using the old version some way to force the commoncode version is required, e.g. as an additional argument here which is ignored if empty.

Reply:
So should we go for the `requirements.txt` approach here, too?

Reply:
Yes, or maybe install the tar.gz in case there could be version conflicts with other tools installed via Pip?

Reply:
> or maybe install the tar.gz

Yes, that's documented as the [recommended method](https://github.com/nexB/scancode-toolkit#installation).

Reply:
However, I currently don't have time to continue working on this.

Reply:
> So should we go for the `requirements.txt` approach here, too?

In view of the [problems I'm facing with installing ScanCode from binaries](https://github.com/oss-review-toolkit/ort/pull/6106#discussion_r1035739831), I went for the proven `requirements.txt` approach (that is successfully used in the "legacy" `Dockerfile`) also in the root `Dockerfile`.